### PR TITLE
fix: reduce surface area for form-associated component types

### PR DESCRIPTION
**Pull request by radium-v**
# Description
This change updates all form-associated components to create a named local class before applying the `FormAssociated` mixin function.

## Motivation & context

It turns out this solves a lot of the `import()` problems that are preventing us from upgrading the API Extractor and TypeScript 4. It also reduces the opportunities for collisions as mitigated in #4475,

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [x] **Bug fix**: A change that fixes an issue, link to the issue above.
- [ ] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

This change should only affect the generated `.d.ts` files.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: packages/web-components/fast-foundation/src/checkbox/checkbox.ts [hunk 1]
```diff
@@ -115,6 +115,9 @@ export class Checkbox extends FormAssociatedCheckbox {
     constructor() {
         super();
 
+        this.defaultChecked = !!this.checkedAttribute;
```
Comment (EisenbergEffect) [inline packages/web-components/fast-foundation/src/checkbox/checkbox.ts:118]: Is this change also related to the TS upgrade?

**Review by EisenbergEffect — APPROVED** [review]

File: packages/web-components/fast-foundation/src/checkbox/checkbox.ts (hunk 1 above)
Comment (radium-v) [reply packages/web-components/fast-foundation/src/checkbox/checkbox.ts:118]: Yes, the transpiled output is functionally equivalent even though TS 4 has stricter checks for `TS2729`:

```
src/checkbox/checkbox.ts:61:45 - error TS2729: Property 'checkedAttribute' is used before its initialization.

61     public defaultChecked: boolean = !!this.checkedAttribute;
                                               ~~~~~~~~~~~~~~~~
```

**Review by nicholasrice — APPROVED** [review]

**Review by chrisdholt — APPROVED** [review]
